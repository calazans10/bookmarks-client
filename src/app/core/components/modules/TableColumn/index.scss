@import 'styles/utilities';

.table-column {
  display: block;

  &:before {
    display: block;
    margin-bottom: rem(10);
    font-size: em(18);
    font-weight: 500;
    content: attr(data-label);
  }

  &:not(:last-child) {
    margin-bottom: rem(20);
  }

  @media(min-width: 600px) {
    display: table-cell;
    padding: rem(12);

    &:before {
      display: none;
    }

    &:not(:last-child) {
      margin-bottom: 0;
    }

    &--actions {
      width: 40%;
      text-align: right;
    }
  }
}
