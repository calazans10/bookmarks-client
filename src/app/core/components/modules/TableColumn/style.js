import styled, { css } from 'styled-components';
import { em, rem } from 'polished';

export const Container = styled.td`
  display: block;

  &:before {
    display: block;
    margin-bottom: ${rem('10px')};
    font-size: ${em('18px')};
    font-weight: 500;
    content: attr(data-label);
  }

  &:not(:last-child) {
    margin-bottom: ${rem('20px')};
  }

  @media (min-width: 600px) {
    display: table-cell;
    padding: ${rem('12px')};

    &:before {
      display: none;
    }

    &:not(:last-child) {
      margin-bottom: 0;
    }

    ${props =>
      props.hasActions &&
      css`
        width: 40%;
        text-align: right;
      `}
  }
`;
