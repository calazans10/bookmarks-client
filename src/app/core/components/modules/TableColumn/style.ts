import styled, { css } from 'styled-components';
import { em, rem } from 'polished';

type ContainerProps = {
  hasActions: boolean;
};

export const Container = styled.td<ContainerProps>`
  display: block;

  &:before {
    content: attr(data-label);
    display: block;
    margin-bottom: ${rem('10px')};
    font-size: ${em('18px')};
    font-weight: 500;
  }

  &:not(:last-child) {
    margin-bottom: ${rem('20px')};
  }

  @media (min-width: 600px) {
    display: table-cell;
    padding: ${rem('12px')};

    &:before {
      display: none;
    }

    &:not(:last-child) {
      margin-bottom: 0;
    }

    ${props =>
      props.hasActions &&
      css`
        width: 40%;
        text-align: right;
      `}
  }
`;
