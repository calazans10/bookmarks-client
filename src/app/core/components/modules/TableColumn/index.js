import React from 'react';
import PropTypes from 'prop-types';
import { Container } from './style';

const TableColumn = ({ label, hasActions, children }) => (
  <Container hasActions={hasActions} data-label={label}>
    {children}
  </Container>
);

TableColumn.propTypes = {
  label: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  hasActions: PropTypes.bool,
};

TableColumn.defaultProps = {
  hasActions: false,
};

export default TableColumn;
