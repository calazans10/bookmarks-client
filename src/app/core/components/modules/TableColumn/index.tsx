import React from 'react';
import { Container } from './style';

type TableColumnProps = {
  label: string;
  hasActions?: boolean;
  children: React.ReactNode;
};

const TableColumn = ({ label, hasActions = false, children }: TableColumnProps) => (
  <Container hasActions={hasActions} data-label={label}>
    {children}
  </Container>
);

export default TableColumn;
