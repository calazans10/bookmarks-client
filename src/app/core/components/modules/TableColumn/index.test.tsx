import React from 'react';
import { render } from '@testing-library/react';
import TableColumn from './index';

describe('TableColumn', () => {
  it('renders without crashing', () => {
    const tableRow = document.createElement('tr');
    const { container } = render(<TableColumn label="Name">John Doe</TableColumn>, {
      container: document.body.appendChild(tableRow),
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
