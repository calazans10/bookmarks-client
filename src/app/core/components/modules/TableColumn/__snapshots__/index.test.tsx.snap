// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableColumn renders without crashing 1`] = `
<td
  class="sc-AxjAm ePbalm"
  data-label="Name"
>
  John Doe
</td>
`;
